src/root.ml

```diff
@@ -362,6 +362,15 @@ module Make (B : Context.A_DISK) = struct
     in
     let previous_generation = Sector.force_id t.current in
     let* queue =
+      let old_generation = 4 in
+      if Int64.to_int t.generation > old_generation
+      then (
+        let gen = B.Id.of_int (Int64.to_int B.nb_sectors + 1) in
+        Queue.pop_old_generation queue (gen, 1))
+      else Lwt_result.return queue
+    in
+    let* queue =
+      let* queue = Queue.push_back queue [ B.Id.of_int (Int64.to_int B.nb_sectors + 1), 1 ] in
```

Comment:
Nice! If you are looking for other constants to delimit the generations, I believe there are other impossible values that could be used as a separator to make it super obvious that this is not a valid element of the queue (... eg with a sector id in the reserved root area and/or a weird range length)